### PR TITLE
Introduce `rz_core_gadget_print()` API to print visual gadgets

### PR DESCRIPTION
**Pull request by officialcjunior**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Introduced `rz_core_gadget_print()` for the command `pg` which displays the gadgets on the screen and used it everywhere `rz_core_cmd` call to `pg` was called. 

**Test plan**

Added a test at `test/db/cmd/cmd_visual`. Here's a screenshot of `pg 10 10 10 10 ddr` in action.
Screenshot_2021-05-12_00-03-27

**Closing issues**

Related to #382

### CODE REVIEW

**Comment by XVilka** [comment]
Please address this warning, apart from that LGTM:
```
../librz/core/core.c: In function ‘rz_core_prompt’:
../librz/core/core.c:2812:3: error: implicit declaration of function ‘rz_core_gadget_print’; did you mean ‘rz_core_gadget_free’? [-Werror=implicit-function-declaration]
 2812 |   rz_core_gadget_print(r);
      |   ^~~~~~~~~~~~~~~~~~~~
      |   rz_core_gadget_free
cc1: all warnings being treated as errors
```

File: librz/core/cmd_print.c [hunk 1]
```diff
@@ -1431,6 +1431,18 @@ RZ_API void rz_core_gadget_free(RzCoreGadget *g) {
 	free(g);
 }
```
Comment (XVilka) [inline librz/core/cmd_print.c:1438]: Please add doxygen description, just `\brief` would be enough

File: librz/include/rz_core.h [hunk 2]
```diff
@@ -397,6 +397,7 @@ typedef int (*RzCoreSearchCallback)(RzCore *core, ut64 from, ut8 *buf, int len);
 RZ_API int rz_core_bind(RzCore *core, RzCoreBind *bnd);
```
Comment (XVilka) [inline librz/include/rz_core.h:398]: Maybe a comment here that these are about ROP gadgets

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
Regarding tests - there are some visual tests already, you could check and add a test for this as well.

**Comment by officialcjunior** [comment]
> Regarding tests - there are some visual tests already, you could check and add a test for this as well.

Are you referring to the ones in `test/db/cmd/cmd_visual`? Or are there any other visual-related tests at someplace else?

**Comment by XVilka** [comment]
Yes, I was referring to this one.

File: librz/include/rz_core.h (hunk 2 above)
Comment (officialcjunior) [reply librz/include/rz_core.h:398]: ROP Gadgets? I'm sorry, which one are you referring to? 
Both of them are related to the these print gadgets, right?

Comment (XVilka) [reply librz/include/rz_core.h:398]: Yes, ignore my comment. It has nothing to do with ROP - I checked the code and it's just a "print gadgets". Probably worth documenting what exactly it is - "printing a result of some command at the specified screen position" or something like that.

**Comment by XVilka** [comment]
Also, one more request - since you are on it, could you please port `pg` to the newshell as well?

**Review by XVilka — APPROVED** [review]

**Pull request merged**